hack/gen-openapi-spec/main.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/go-openapi/spec"
+	tfjobspec "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1"
+	"k8s.io/kube-openapi/pkg/common"
+)
+
+// Generate OpenAPI spec definitions for KFService Resource
+func main() {
+	if len(os.Args) <= 1 {
+		log.Fatal("Supply a version")
+	}
+	version := os.Args[1]
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+	oAPIDefs := tfjobspec.GetOpenAPIDefinitions(func(name string) spec.Ref {
+		return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name)))
+	})
+	defs := spec.Definitions{}
+	for defName, val := range oAPIDefs {
+		defs[swaggify(defName)] = val.Schema
+	}
+	swagger := spec.Swagger{
+		SwaggerProps: spec.SwaggerProps{
+			Swagger:     "2.0",
+			Definitions: defs,
+			Paths:       &spec.Paths{Paths: map[string]spec.PathItem{}},
+			Info: &spec.Info{
+				InfoProps: spec.InfoProps{
+					Title:   "TFOperator",
```

Comment:
Can add Description here for swagger-codegen, so avoid "No description provided" in all files.